crates/sui-indexer/src/handlers/tx_processor.rs

```diff
@@ -289,9 +289,20 @@ pub(crate) struct EpochEndIndexingObjectStore<'a> {
 
 impl<'a> EpochEndIndexingObjectStore<'a> {
     pub fn new(data: &'a CheckpointData) -> Self {
-        // We only care about output objects for end-of-epoch indexing
+        // We only care about latest output objects at the end of an epoch
+        let mut output_objects_by_id = BTreeMap::new();
+        for obj in data.output_objects() {
```

Comment:
I think to reduce the chance of having similar mistakes, we should rename and change the semantics of output_objects to always return the latest object state, and also handle objects that are deleted in it.
There should be only one direct use of the current output_objects(), which we should just replace with a iter on transactions.

Another question: do we need backfill on this?

Reply:
sg re the output_objects function change.

re backfill, I was planning to run a query to see if we have multiple mutations of the system object in one end-of-epoch checkpoint, if so ideally I will figure a way to fix it in place as it's very small change in DB and now backfill from genesis is super slow.

Reply:
We may need to backfill anyway due to the other object related bugs.

Reply:
Alternatively, if it becomes too much to fix this together with other object related bugs (especially around the call to output_objects()), an easy walkaround just for this issue is to change the iter() to rev() in the get_object impl

Reply:
rev would do indeed, but the semantics of `objects` of `EpochEndIndexingObjectStore` still fails as some can be intermediate instead of exact end of epoch. I think changing output objects to only have latest live objects is a good idea.